### PR TITLE
Count support in query

### PR DESCRIPTION
**Pull request by howawong**
- Add overallCount to Query and RecordQueryRequest
- Add QueryInfo class
- Add extra method to RecordQueryResponseHandler
- Update unit test
- Update example activity

Connects #211

### CODE REVIEW

File: skygear/src/main/java/io/skygear/skygear/RecordQueryResponseHandler.java [hunk 1]
```diff
@@ -50,7 +57,17 @@ public void onSuccess(JSONObject result) {
...
+                JSONObject infoJson = result.getJSONObject("info");
+                Integer overallCount = null;
+                if (infoJson.has("count")) {
+                    overallCount = infoJson.getInt("count");
+                }
+                QueryInfo queryInfo = new QueryInfo(overallCount);
+                this.onQuerySuccess(records, queryInfo);
+            } else {
+                this.onQuerySuccess(records);
+            }
```
Comment (carmenlau) [inline skygear/src/main/java/io/skygear/skygear/RecordQueryResponseHandler.java]: I think should call both `onQuerySuccess` for all case, user will be confused which callback should they use and not aware why their `onQuerySuccess` won't call if they choose the wrong one

**Comment by carmenlau** [comment]
@howawong Update test case please

File: skygear/src/androidTest/java/io/skygear/skygear/RecordQueryResponseHandlerUnitTest.java [hunk 2]
```diff
@@ -113,6 +117,12 @@ public void onQuerySuccess(Record[] records) {
                 checkpoints[0] = true;
             }
 
+            @Override
+            public void onQuerySuccess(Record[] records, QueryInfo info) {
+                assertEquals(3, info.getOverallCount().intValue());
+                checkpoints[0] = true;
```
Comment (carmenlau) [inline skygear/src/androidTest/java/io/skygear/skygear/RecordQueryResponseHandlerUnitTest.java]: I think the `checkpoints` propose is to make sure the function is called. We did 2 tests here for 2 `onQuerySuccess` function, if we set both set `checkpoints[0] = true`, we will not know if one of them is not called.

Comment (howawong) [reply skygear/src/androidTest/java/io/skygear/skygear/RecordQueryResponseHandlerUnitTest.java]: updated.

Comment (carmenlau) [reply skygear/src/androidTest/java/io/skygear/skygear/RecordQueryResponseHandlerUnitTest.java]: @howawong Have you pushed your changes? The code looks the same?

**Pull request merged**